### DISCUSSION TITLE
Nested fieldArray items are not persisting between insertions

### DISCUSSION THREAD

**Discussion by niksmac** [react-hook-form/react-hook-form#4025] [Q&A]
Code Sandbox  https://codesandbox.io/s/interesting-robinson-hehc8 

Step to reproduce: 

  1. Choose hobbies from Add information 
  2. Click Add more
  3. Then Click Prepend

Expected behaviour:
  1. The initial items should maintain the hobbies added 

Current behaviour:
  1. Losing options added (nested field)

**Comment by bluebill1049** [comment]
because the `defaultValues` from the `fields` object is not getting updated (the entire field array is not getting updated) solutions: 

1. toggle the CSS visibility of the inputs

2. set shouldUnregister: false;  and watch the entire field array to get the latest value.
https://codesandbox.io/s/fast-sound-ugbke?file=/src/App.js
<img width="846" alt="Screen Shot 2021-01-26 at 6 28 28 pm" src="https://user-images.githubusercontent.com/10513364/105813668-4ba25e80-6004-11eb-9458-4cb2142f7b8e.png">

**Reply by bluebill1049** [reply]
also have a look at this example: https://github.com/react-hook-form/react-hook-form/blob/master/app/src/UseFieldArrayUnregister.tsx
